src/probspecs.nim

```diff
@@ -100,13 +102,70 @@ proc findProbSpecsExercises(repo: ProbSpecsRepo, conf: Conf): seq[ProbSpecsExerc
     if conf.exercise.isNone or conf.exercise.get() == repoExercise.slug:
       result.add(initProbSpecsExercise(repoExercise))
 
-proc findProbSpecsExercises*(conf: Conf): seq[ProbSpecsExercise] =
-  let probSpecsRepo = initProbSpecsRepo()
-
+template withDir(dir: string; body: untyped): untyped =
+  ## Changes the current directory to `dir` temporarily.
+  let startDir = getCurrentDir()
   try:
-    probSpecsRepo.remove()
-    probSpecsRepo.clone()
-    probSpecsRepo.grainsWorkaround()
-    probSpecsRepo.findProbSpecsExercises(conf)
+    setCurrentDir(dir)
+    body
   finally:
-    probSpecsRepo.remove()
+    setCurrentDir(startDir)
+
+proc findProbSpecsExercises*(conf: Conf): seq[ProbSpecsExercise] =
+  const mainBranchName = "master"
+
+  if conf.probSpecsDir.isSome():
+    let probSpecsDir = conf.probSpecsDir.get()
+    logDetailed(&"Using user-provided problem-specifications dir: {probSpecsDir}")
+
+    if not dirExists(probSpecsDir):
+      showError(&"the given problem-specifications directory does not exist: '{probSpecsDir}'")
+
+    withDir probSpecsDir:
+      if execCmd("git rev-parse") != 0:
+        showError(&"the given problem-specifications directory is not a git repository: '{probSpecsDir}'")
+
+      # Exit if the working directory is not clean.
+      if execCmd("git diff-index --quiet HEAD") != 0: # Ignores untracked files.
+        echo &"\nUnstaged changes in {probSpecsDir}:"
+        discard execCmd("git status --short --untracked=no")
+        showError(&"the given problem-specifications working directory is not clean: '{probSpecsDir}'")
```

Comment:
Maybe output this the other way around?

```suggestion
        showError(&"the given problem-specifications working directory has unstaged changes: '{probSpecsDir}'")
        discard execCmd("git status --short --untracked=no")
```

Reply:
Doesn't work, unfortunately. `showError` isn't the best name, but it causes an exit.

But I agree that it might be better to see the `git status` output after the error, rather than before.

Reply:
An alternative would be to not display the unstaged changes.

Reply:
Done.